### PR TITLE
check plan size in plan_execution

### PR DESCRIPTION
**Pull request by johaq**
see #490

### CODE REVIEW

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 1]
```diff
@@ -477,18 +477,22 @@ void plan_execution::PlanExecution::doneWithTrajectoryExecution(
...
-
   // if any side-effects are associated to the trajectory part that just completed, execute them
-  if (plan->plan_components_[index].effect_on_success_)
-    if (!plan->plan_components_[index].effect_on_success_(plan))
-    {
-      // execution of side-effect failed
-      ROS_ERROR("Execution of path-completion side-effect failed. Preempting.");
-      preempt_requested_ = true;
-      return;
+    if(plan->plan_components_.size() != 0) {
```
Comment (v4hn) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: Could you please change the logic to
```
if(plan->plan_components_.size() == 0){
  ... report warning ...
  return;
}
```
?
Multiple MoveIt developers agreed that this is the preferred style in changed MoveIt code.

**Comment by v4hn** [comment]
Also the format check failed due to wrong indention.

**Comment by davetcoleman** [comment]
Thank you this PR! I agree with @v4hn

**Comment by v4hn** [comment]
Thanks for adjusting the request!
Format checking failed again though (I know it's annoying...), could you please cleanup the whitespace changes?

**Comment by johaq** [comment]
Cleaned it up. I hope everything is fine now.

**Comment by davetcoleman** [comment]
There is still a code format error, see the [log](https://travis-ci.org/ros-planning/moveit/jobs/227425057#L188) where it has +/- in front of lines of incorrectly indented and spaced code.

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 2]
```diff
@@ -477,9 +477,14 @@ void plan_execution::PlanExecution::doneWithTrajectoryExecution(
 void plan_execution::PlanExecution::successfulTrajectorySegmentExecution(const ExecutableMotionPlan* plan,
                                                                          std::size_t index)
 {
-  ROS_DEBUG("Completed '%s'", plan->plan_components_[index].description_.c_str());
+  if (plan->plan_components_.size() == 0)
+  {
+    ROS_WARN("Length of provided motion plan is zero.");
```
Comment (davetcoleman) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: can you make these two ``ROS_`` messages have a namespace, also?

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 3]
```diff
@@ -522,7 +522,10 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
 
   bool setPoseTargets(const std::vector<geometry_msgs::PoseStamped>& poses, const std::string& end_effector_link)
   {
+      ROS_ERROR_STREAM_NAMED("move_group_interface","end_effector_link_" << end_effector_link_ << "\n");
+      ROS_ERROR_STREAM_NAMED("move_group_interface","end_effector_link" << end_effector_link << "\n");
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: extra indentation here. also, the "\n" will cause double line breaks, which is not necessary.

also - the line is repeated twice?

**Comment by davetcoleman** [comment]
There are still some formatting issues that Travis requires to be fixed, thanks

**Comment by v4hn** [comment]
superseded by #492, so I will close the request.
Thanks @johaq

**Pull request closed without merging**